src/query/functions/src/scalars/bitmap.rs

```diff
@@ -143,6 +144,38 @@ pub fn register(registry: &mut FunctionRegistry) {
         ),
     );
 
+    registry.register_passthrough_nullable_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType, _, _>(
+        "sub_bitmap",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType>(
+            |b, offset, length, builder, ctx| {
+                    match RoaringTreemap::deserialize_from(b) {
+                        Ok(rb) => {
+                            let subset_start = offset;
+                            let subset_length = length;
+                            if subset_start >= b.len() as u64 {
+                                let mut rb = RoaringTreemap::new();
+                                rb.serialize_into(&mut builder.data).unwrap();
+                                builder.commit_row();
+                            } else {
+                                let adjusted_length = (subset_start + subset_length).min(b.len() as u64) - subset_start;
+                                let subset_bitmap = &rb.into_iter().collect::<Vec<_>>()[subset_start as usize ..(subset_start + adjusted_length) as usize];
+                                let mut rb = RoaringTreemap::new();
+                                for a in RoaringTreemap::from_iter(subset_bitmap.iter()) {
```

Comment:
using `RoaringTreemap::from_iter` can build a `RoaringTreemap`, so we can remove `RoaringTreemap::new` and for loop insert.